OSDOCS-12223: Documented the 4.16.16 z-stream notes (#82970)
openshift/openshift-docs: release_notes/ocp-4-16-release-notes.adoc

For any {product-title} release, always review the instructions on xref:../updating/updating_a_cluster/updating-cluster-web-console.adoc#updating-cluster-web-console[updating your cluster] properly.
====

// 4.16.16
[id="ocp-4-16-16_{context}"]
=== RHSA-2024:7599 - {product-title} {product-version}.16 bug fix and security update

Issued: 09 October 2024

{product-title} release {product-version}.16 is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2024:7599[RHSA-2024:7599] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHBA-2024:7602[RHBA-2024:7602] advisory.

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.16.16 --pullspecs
----

[id="ocp-4-16-16-bug-fixes_{context}"]
==== Bug fixes

> * Previously, the `metal3-ironic-inspector` container/pod in the `openshift-machine-api` namespace caused memory consumptions issues for your cluster. With this release, the memory consumption issue is fixed so this issue no loner persists. (link:https://issues.redhat.com/browse/OCPBUGS-42113[*OCPBUGS-42113*])

Reviewer: ```suggestion
* Previously, the `metal3-ironic-inspector` container/pod in the `openshift-machine-api` namespace caused memory consumption issues for your cluster. With this release, the memory consumption issue is fixed so this issue no loner persists. (link:https://issues.redhat.com/browse/OCPBUGS-42113[*OCPBUGS-42113*])
```

> * Previously, the `metal3-ironic-inspector` container/pod in the `openshift-machine-api` namespace caused memory consumptions issues for your cluster. With this release, the memory consumption issue is fixed so this issue no loner persists. (link:https://issues.redhat.com/browse/OCPBUGS-42113[*OCPBUGS-42113*])

Reviewer: Should this be generalized to "for clusters"?
```suggestion
* Previously, the `metal3-ironic-inspector` container/pod in the `openshift-machine-api` namespace caused memory consumptions issues for clusters. With this release, the memory consumption issue is fixed so this issue no loner persists. (link:https://issues.redhat.com/browse/OCPBUGS-42113[*OCPBUGS-42113*])
```

> * Previously, the `metal3-ironic-inspector` container/pod in the `openshift-machine-api` namespace caused memory consumptions issues for your cluster. With this release, the memory consumption issue is fixed so this issue no loner persists. (link:https://issues.redhat.com/browse/OCPBUGS-42113[*OCPBUGS-42113*])

Reviewer: "longer" is misspelled but honestly I think the end of that sentence can be removed anyways:

```suggestion
* Previously, the `metal3-ironic-inspector` container/pod in the `openshift-machine-api` namespace caused memory consumptions issues for your cluster. With this release, the memory consumption issue is fixed. (link:https://issues.redhat.com/browse/OCPBUGS-42113[*OCPBUGS-42113*])
```

> * Previously, creating cron jobs to create pods for your cluster caused the component that fetches the pods to fail. Because of this issue, the *Topology* page on the {product-title} web console to fail. With this release, a `3` second delay is configured for the component that fetches pods that are generated from the cron job so that this issue no longer exists. (link:https://issues.redhat.com/browse/OCPBUGS-42015[*OCPBUGS-42015*])

Reviewer: ```suggestion
* Previously, creating cron jobs to create pods for your cluster caused the component that fetches the pods to fail. Because of this issue, the *Topology* page on the {product-title} web console failed. With this release, a `3` second delay is configured for the component that fetches pods that are generated from the cron job so that this issue no longer exists. (link:https://issues.redhat.com/browse/OCPBUGS-42015[*OCPBUGS-42015*])
```

* Previously, because of an internal bug, the Node Tuning Operator incorrectly computed CPU masks for interrupt and network handling CPU affinity if a machine had more than 256 CPUs. This prevented proper CPU isolation on those machines and resulted in `systemd` unit failures. With this release, the Node Tuning Operator computes the masks correctly. (link:https://issues.redhat.com/browse/OCPBUGS-39377[*OCPBUGS-39377*])

> * Previously, when you used the Redfish Virtual Media to add an xFusion bare-metal node to your cluster, the node did not get added because of a node registration issue. The issue occurred because the hardware was not 100 percent compliant with Redfish. With this release, you can now add xFusion bare-metal nodes to your cluster.(link:https://issues.redhat.com/browse/38797[*38797*]) 

Reviewer: This link needs to be fixed
